TemporaryParser/TPRBParser.class.st

```diff
@@ -704,28 +717,30 @@ TPRBParser >> parseNumberLiteral [
 
 { #category : #'private-parsing' }
 TPRBParser >> parseParenthesizedExpression [
-	| leftParen node |
-	leftParen := currentToken start.
+	| node startToken |
+	startToken := currentToken.
 	self step.
 	node := self parseAssignment.
 	^(currentToken isSpecial and: [currentToken value = $)])
 		ifTrue: 
-			[node addParenthesis: (leftParen to: currentToken start).
+			[node addParenthesis: (startToken start to: currentToken start).
 			self step.
 			node]
-		ifFalse: [ (self parserError: ''')'' expected') start: leftParen]
+		ifFalse: [ ^errorBlock ifNil: [ self parserError: ''')'' expected']
+					  				  ifNotNil: [ self parseEnglobingError: (OrderedCollection with: node) with: startToken]]
 ]
 
 { #category : #'private-parsing' }
 TPRBParser >> parsePragma [
 	"Method called by parsePragmas which already verified that current token is the special character <"
-	| start pragma |
-	start := currentToken start.
+	| startToken pragma |
+	startToken := currentToken.
 	self step.
 	pragma := self basicParsePragma.
 	(currentToken isBinary and: [ currentToken value == #> ]) 
-		ifFalse: [ ^ self addPragma: (self parserError: '''>'' expected') ].
-	pragma left: start; right: currentToken start.
+		ifFalse: [  ^self addPragma: (errorBlock ifNil: [ self parserError: '''>'' expected']
+					  						 ifNotNil: [ self parseEnglobingError: (OrderedCollection with: pragma) with: startToken])].

```

Comment:
I see this `errorBlock ifNil: [...] ifnotNil: [ self parseEnglobingError: ]` many times already. We could refactor it a bit to avoid duplications